### DISCUSSION TITLE
Rebuild of a MC roaster. Electronic Control

### DISCUSSION THREAD

**Discussion by ContrastCoffee** [artisan-roaster-scope/artisan#915] [Q&A]
I've got a 30KG MC roaster we have been rebuilding.  So far, new probes, new gas lines, new burners, new exhaust and now we are working on digital control from Artisan for gas.  (after this we will be working on fan speed controls).
2 different thoughts on changing the manual gas control to digital using artisan.  We already have Phidgets working on the machine for this.  We have an electric ball valve hooked up to artisan and we are able to control gas quite well.  The issue is the accuracy, especially on the low end of the gas settings is not good at all.  We can set Artisan to 30% gas and we will get .4 bars kPa if we are coming down to the 30%.  If we try to come up to 30% from 20% or something, we will get .2 bars kPa. This then creates some issues with profiling and default profiles.  I really don't want to have to look at the analog gas gauge to set pressures, I would rather just have what we are doing in Artisan be accurate enough with the percentages.
2 different ways I think I could go about this...
1.) Buy a very high end electric ball valve that doesn't have any "slop" in it.  I am pretty certain the difference we are seeing when coming up or down is due to a little slop in the valve.  Cost around $600.
2.) I have a digital gas pressure gauge.  Question here is can Artisan read this gauge and then we somehow have Artisan use the pressure reading rather than setting the valve with a percentage???  This I feel would be the most accurate, but may not be feasible.

Thoughts???

**Comment by poundy** [comment]
I'm far from an expert here. This is going to be a detailed PID discussion I think but I'll start the conversation... 

Gas measurement and control is always a touchy subject ( 🤣 ) because at low pressures, responses are almost always non-linear. It really means that for accuracy, you need to purchase appropriate control valves for the task at hand.  

Having said that, the automation you are looking at doing will be important, as what you actually will end up measuring is the BT tracking the background, not the gas pressure changes - the pressure is a way to influence the BT change. If you need twice the pressure to stay on target BT, then that matters less than the fact that the ball valve is set to .2 bar or .4bar. What will matter is that the PID will want to change it and have as small an error on that as possible, so slop will not be helpful.

As for your pressure gauge, there are always ways of connecting it to Artisan, but without any info on your device there's not much to say - but again, it actually is just another data point, and if the PID isn't linked to it then there's not much value in that data in my opinion.

**Reply by ContrastCoffee** [reply]
Poundy, thanks for kicking off the responses.

At this point I am not looking at automation, but rather just digital control.  We have always used manual control on the gas valve to adjust to a certain pressure and then made the notation using sliders or buttons in Artisan.  I now have the slider in Artisan controlling the gas valve, but like I said, it is not accurate to pressure.  So, if I can setup the digital pressure gauge in Artisan and have the slider adjust to that pressure when we are using the slider, this would be ideal.  Currently the slider only goes by percentage and is not accurate to the actual gas pressure consistently.

If I'm reading you correctly the better fix may just be a very good motorized ball valve with no slop, thus giving better consistency when we use the slider to adjust the pressure?

**Comment by ContrastCoffee** [comment]
bump
